week2/Program.cs

```diff
@@ -0,0 +1,183 @@
+    foreach (char c in str)
+        if (!Char.IsWhiteSpace(c))
+        {
+            characters.Add(c);
+        }
+    return characters;
+};
+
+app.MapGet("/prep-task3", (string str) =>
+{
+    List<char> characterList = CharacterList(str);
+    char[] vowels = { 'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U' };
+    int vowelCount = 0;
+    foreach (char c in characterList)
+    {
+        if (vowels.Contains(c)) vowelCount++;
+    }
+    return vowelCount;
+});
+
+app.MapGet("/prep-task4", (string str) =>
+{
+    List<char> characterList = CharacterList(str);
+    int firstOccurenceOfS = characterList.FindIndex(x => x == 's');
+    int lastOccurenceOfE = characterList.FindLastIndex(x => x == 'e');
+    return Results.Ok(new { FirstOccurenceOfS = firstOccurenceOfS, LastOccurenceOfE = lastOccurenceOfE }); ;
+});
+
+app.MapGet("/prep-task5", () => NumbersToHundredList());
+
+List<int> NumbersToHundredList()
+{
+    List<int> numbers = new List<int>();
+    for (int i = 0; i <= 100; i++)
+    {
+        numbers.Add(i);
+    }
+    return numbers;
+};
+
+app.MapGet("/prep-task6", () =>
+{
+    List<int> numbersToHundredList = NumbersToHundredList();
+    numbersToHundredList.RemoveRange(33, 12);
+    //numbersToHundredList.ForEach(Console.WriteLine);
+    return numbersToHundredList;
+});
+
+app.MapGet("/prep-task7", () =>
+{
+    // I did only 1st part of the 7th exercise - to split the list.
+    // But couldn't proceed to next as the return type here is in string.
+    // I tried Chunk and GroupBY to split the list and its return is in string.
+    List<int> numbersToHundredList = NumbersToHundredList();
+    int numberOfElementsInSplittedList = (numbersToHundredList.Count + 1) / 2;
+
+    string splittedList = System.Text.Json.JsonSerializer.Serialize(numbersToHundredList.Chunk(numberOfElementsInSplittedList));
+    return splittedList;
+
+    // var splittedList = numbersToHundredList.Select((x, idx) => new { index = idx, val = x })
+    //         .GroupBy(i => i.index / numberOfElementsInSplittedList)
+    //         .Select(g => g.Select(x => x.val));
+    // return splittedList;
+});
+
+// Class exercises
+app.MapGet("/class-task1", (string number1, string number2, string operation) =>
+{
+    bool numberOneIsValid = int.TryParse(number1, out int num1);
+    bool numberTwoIsValid = int.TryParse(number2, out int num2);
+    int result = 0;
+    if (!numberOneIsValid && !numberTwoIsValid)
+    {
+        return Results.BadRequest("Not a valid number");
+    }
+    switch (operation)
+    {
+        case "add":
+            result = num1 + num2;
+            break;
+        case "substract":
+            result = num1 - num2;
+            break;
+        case "multiply":
+            result = num1 * num2;
+            break;
+        default:
+            return Results.BadRequest("Not a valid operation");
+    }
+    return Results.Ok(result);
+
+});
+
+app.MapGet("/class-task2", (string input) =>

```

Comment:
You would usually use methods with [expression body](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/lambda-operator#expression-body-definition) (think it as an arrow function from JS) when you have one line of code that you want to call.

So, for a block of the code, you would use standard brackets:

```
app.MapGet("/class-task2", (string input)
{
  // block (few lines of code) of code here
}
```

and if you have one line of code it can be done with body statement.
```
app.MapGet("/alphabetical", (string input) => GetUniqueCharacters(input)
```

This is of course a style recommendation and your code will work perfectly without applying it.